### ISSUE TITLE
Use sciml_train for multiple initial conditions

### ISSUE THREAD

**Issue by jiweiqi** [SciML/DiffEqFlux.jl#306]
Hi,

There is a tutorial on mini-batch for a single initial condition https://diffeqflux.sciml.ai/dev/examples/minibatch/.

However, what I am going to do is to train neural ode for multiple initial conditions. And I want to use sort of mini-batching by update the neural network with only one initial condition each time.

I tried something like
```Julia
n_exp = 4
u0_list = rand(Float64, (n_exp, ns))
i_exp = 1

function predict_neuralode(p)
    global i_exp
    i_exp = rand(1:n_exp)
    u0 = u0_list[i_exp, :]
    return clamp.(Array(prob_neuralode(u0, p)), -ub, ub)
end

function loss_neuralode(p)
    pred = predict_neuralode(p)
    loss = sum(abs2, ode_data_list[i_exp] .- pred)
    return loss, pred
end

pstart = DiffEqFlux.sciml_train(loss_neuralode, p, ADAM(0.001), cb = cb, maxiters = 10000).minimizer
```

But the line of ` i_exp = rand(1:n_exp)` inside `function predict_neuralode(p)` is not working, throwing a tons of errors. Maybe the error is from the way I define global variable `i_exp`?

**Comment by jiweiqi** [comment]
For now, I can do the task by compute the gradient manually without using `sciml_train`.

**Comment by ChrisRackauckas** [comment]
Zygote doesn't seem to like that rand. I think you need to define an `@nograd take_rand(n_exp) = rand(1:n_exp)` and use that.
